### PR TITLE
Set product info in DI instead of Rest Client

### PR DESCRIPTION
**Pull request by Y-Sindo**

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs [hunk 1]
```diff
@@ -221,6 +230,10 @@ private static IServiceCollection AddRestClientFactory(this IServiceCollection s
             static bool IsTransientErrorForNonMessageApi(HttpStatusCode code) =>
                 code >= HttpStatusCode.InternalServerError ||
                 code == HttpStatusCode.RequestTimeout;
+
+            static void ConfigureProduceInfo(IServiceProvider sp, HttpClient client) =>
+                client.DefaultRequestHeaders.Add(Constants.AsrsUserAgent, sp.GetRequiredService<IOptions<ServiceManagerOptions>>().Value.ProductInfo);
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs:235]: could ProductInfo be null?

Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs:235]: No

Comment (vicancy) [reply src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs:235]: Give it a default value if it is null?

Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs]: Fixed.

**Review by vicancy — APPROVED** [review]

**Pull request merged**